code/modules/awaymissions/maploader/writer.dm

```diff
@@ -31,40 +32,63 @@ dmm_suite{
 		if(!isturf(t1) || !isturf(t2)){
 			CRASH("Invalid arguments supplied to proc save_map, arguments were not turfs.")
 			}
-		var/file_text = write_map(t1,t2,flags)
-		if(fexists("[map_name].dmm")){
-			fdel("[map_name].dmm")
+		var/map_prefix = "_maps/quicksave/"
+		var/map_path = "[map_prefix][map_name].dmm"
+		if(fexists(map_path)){
+			fdel(map_path)
 			}
-		var/saved_map = file("[map_name].dmm")
-		saved_map << file_text
+		var/saved_map = file(map_path) // We give the map writer the file directly
+		// Because repeated string appending is super murder for performance
+		var/map_text = write_map(t1,t2,flags,saved_map)
+		saved_map << map_text
 		return saved_map
 		}
-	write_map(var/turf/t1 as turf, var/turf/t2 as turf, var/flags as num){
+	write_map(var/turf/t1 as turf, var/turf/t2 as turf, var/flags as num, var/map_file as file){
 		//Check for valid turfs.
 		if(!isturf(t1) || !isturf(t2)){
 			CRASH("Invalid arguments supplied to proc write_map, arguments were not turfs.")
 			}
-		var/turf/nw = locate(min(t1.x,t2.x),max(t1.y,t2.y),min(t1.z,t2.z))
-		var/turf/se = locate(max(t1.x,t2.x),min(t1.y,t2.y),max(t1.z,t2.z))
+		var/turf/ne = locate(max(t1.x,t2.x),max(t1.y,t2.y),max(t1.z,t2.z)) // Outer corner
+		var/turf/sw = locate(min(t1.x,t2.x),min(t1.y,t2.y),min(t1.z,t2.z)) // Inner corner
 		var/list/templates[0]
 		var/template_buffer = {""}
+		var/buffer_line = {""}
 		var/dmm_text = {""}
-		for(var/pos_z in nw.z to se.z){
-			for(var/pos_y in nw.y to se.y){
-				for(var/pos_x in nw.x to se.x){
+
+
+		// // Don't write more than 1000 maps at once ;)
+		// var/static/cache_index = 0
+		// if(cache_index > 1000)
+		// 	cache_index = 0
+		// cache_index++
+		// var/cache_prefix = "data/map_cache/"
+		// var/cache_name = "mapcache[cache_index].txt"
+		// var/cache_path = "[cache_prefix][cache_name]"
+		// if(fexists(cache_path))
+		// 	fdel(cache_path)
+		// // Here's hoping file writes are less pricey than repeated string appends
+		// var/template_buffer_cache = file(cache_path)
+
```

Comment:
Oh yeah here's an experiment trying to use a file to speed up repeated appending.
That definitely did not work.